### DISCUSSION TITLE
More API information on mdmcert.download

### DISCUSSION THREAD

**Discussion by ben221199** [micromdm/micromdm#971] [General]
Hello all,

A while ago, I came across https://mdmcert.download and this site was literally a dream coming true. I was able to test Apple's APNS endpoint without direclty applying for Apple's Developer Program for $299 a year (business). Now I'm developing my own MDM system, as alternative to MicroMDM and Commandment. However, as seen in the source code of Commandment (and also referenced in the source code of MicroMDM), there is this comment: https://github.com/cmdmnt/commandment/blob/17c1dbe3f5301eab0f950f82608c231c15a3ff43/commandment/apns/mdmcert.py#L19-L24

To respect the owners of both MicroMDM and Commandment, I don't want to use their API key. However, on the site of mdmcert.download there is nothing told about an API and requesting API keys. I also mailed to MacTechs, but I never got any reply. It seems to me that I cannot get my own API key to use for mdmcert.download in this way.

I saw that @jessepeterson worked at MacTechs, so: Jesse, if it is possible, can I receive a API key for my own project? Also, is it possible to update the website to have more information about the API and on how to get an API key for that API?

Thanks in advance and keep up the good work

**Comment by jessepeterson** [comment]
You [no longer need an enterprise account](https://macadmins.slack.com/archives/C5238RU9X/p1686846643735339) for the MDM CSR option. A standard "business" Apple Developer (US$99/yr) account will suffice. 

If you are offering an MDM product for customers you should really be getting your own MDM CSR option and signing certificates on behalf of your customers. The cost is significantly reduced these days.

If you're still interested in an mdmcert.download API key please reach out to me @jessepeterson on the [MacAdmins Slack](https://www.macadmins.org) (we also hang out in #micromdm, #nanomdm, and #mdmdev channels). Thanks!

I'll also add that mdmcert.download is not an officially related project to MicroMDM and we do not provide official support for it from the MicroMDM project. Thanks!

**Reply by ben221199** [reply]
Hi @jessepeterson,

Thank you for your answer. The reason for the $299 is that we want to use the MDM in a company, so money is not the problem. I just don't want to spend the money during developing. After development, it is definitely the plan to get our own MDM CSR.

I think I'm already member of the MacAdmins Slack. I will reach out to you there for the API key. Eventually having some more discussion about it too.

I understand that mdmcert.download is not fully related to MicroMDM. Yes, this dicussion is possible somewhat misplaced. However, for me it seemed the best way to get a initial answer and maybe this can help future developers that need an API key too.
